client/src/main/java/client/Controller.java

```diff
@@ -0,0 +1,292 @@
+                    try {
+                        socket.close();
+                        textArea.appendText("Your connection is closed");
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }).start();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void sendMsg(ActionEvent actionEvent) {
+        try {
+            out.writeUTF(textField.getText());
+            textField.clear();
+            textField.requestFocus();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public void tryToAuth(ActionEvent actionEvent) {
+        authMethod();
+    /*    if (socket == null || socket.isClosed()){
+            connect();
+        }
+        try {
+            out.writeUTF(String.format("%s: %s %s", Commands.AUTH, loginField.getText().trim(), passwordField.getText().trim() ));
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }finally {
+            passwordField.clear();
+        }*/
+
+    }
+
+    public void setTitle(String nickname){
+        Platform.runLater(()->{
+            if (nickname.equals("")){
+                stage.setTitle("Best Chat of World");
+            }else
+                stage.setTitle(String.format("Best Chat of World [%s]",nickname));
+        });
+    }
+
+    public void clientListOnMouseReleased(MouseEvent mouseEvent) {
+        if (mouseEvent.getClickCount() == 2){
+            String msg = String.format("%s, %s ", Commands.PERSONAL_MSG, clintList.getSelectionModel().getSelectedItem());
+            textField.setText(msg);
+        }
+    }
+
+    public void tryToAuthByKey(KeyEvent keyEvent) {
+        if (keyEvent.getCode().equals(KeyCode.ENTER)) {
+            authMethod();
+        }
+    }
+    private void authMethod(){
+            if (socket == null || socket.isClosed()){
+                connect();
+            }
+            try {
+                out.writeUTF(String.format("%s: %s %s", Commands.AUTH, loginField.getText().trim(), passwordField.getText().trim() ));
+
+            } catch (IOException e) {
+                e.printStackTrace();
+            }finally {
+                passwordField.clear();
+            }
+    }
+
+    public void showRegWindow(ActionEvent actionEvent) {
+        if (regStage == null){
+            initRegWindow();
+        }
+        regStage.setTitle("Chat registration");
+        regController.setRegMode(true);
+        regStage.show();
+    }
+    public void changeNick(ActionEvent actionEvent) {
+        if (regStage == null){
+            initRegWindow();
+        }
+        regStage.setTitle("Chat nickname changing");
+        regController.setRegMode(false);
+        regStage.show();
+    }
+
+    private void initRegWindow() {
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("/regManager.fxml"));
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [UI: Usage of GetResource may be unsafe if class is extended (UI_INHERITANCE_UNSAFE_GETRESOURCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ui-usage-of-getresource-may-be-unsafe-if-class-is-extended-ui-inheritance-unsafe-getresource).
Calling this.getClass().getResource(...) could give results other than expected if this class is extended by a class in another package.